tests/main.cpp

```diff
@@ -77,59 +88,253 @@ void parse_args(Module *m, Type *type, int argc, char argv[][40]) {
                       sv->value.f64 = -NAN;
                   } else {
                       sv->value.f64 = atof(argv[i]);
-                  }; break;
+                  }; break;*/
         }
     }
 }
 
-void invoke(Module* m, char* call_f, char args[][40]) {
-    uint32_t fidx =  -1; 
-    m->sp  = -1;
-    m->fp  = -1;
+void invoke(Module *m, const char *call_f, Value values[]) {
+    uint32_t fidx = -1;
+    m->sp = -1;
+    m->fp = -1;
     m->csp = -1;
     //TODO move to the WARDuino class     
     for (uint32_t f = 0; f < m->function_count; f++) {
-	    char *fname = m->functions[f].export_name;
-	    if (!fname) {
-		    continue;
-	    }
-	    if (strncmp(call_f, fname, 1024) == 0) {
-		    fidx = f;
-		    break;
-	    }
+        char *fname = m->functions[f].export_name;
+        if (!fname) {
+            continue;
+        }
+        if (strncmp(call_f, fname, 1024) == 0) {
+            fidx = f;
+            break;
+        }
     }
 
     Block *func = &m->functions[fidx];
-    Type *type  = func->type; 
-    parse_args(m, type, type->param_count, args);
+    Type *type = func->type;
+    parse_args(m, type, type->param_count, values);
     setup_call(m, fidx);
-    interpret(m);
+    interpret(m, true);
+}
 
+void assertValue(Value *val, Module *m) {
```

Comment:
It might be useful to let this function return a boolean such that we can say whether the tests have failed programmatically with the exit code.

Reply:
I'll definitely add that. I've also been thinking of adding an overview at the end, with how many tests failed and how many succeeded.

Reply:
Commit a26cc52 makes assertValue return a bool.